### PR TITLE
feat:add enum

### PR DESCRIPTION
**Pull request by guaguaguaxia**

### CODE REVIEW

File: embedchain/embedchain.py [hunk 1]
```diff
@@ -68,16 +73,20 @@ def _get_chunker(self, data_type):
...
-        if data_type in chunkers:
+        if isinstance(data_type, DataType):
             return chunkers[data_type]
+        # compatible string
+        if not isinstance(data_type, DataType):
+            data_type_enum = DataType.get_enum(data_type)
+            return chunkers[data_type_enum]
         else:
-            raise ValueError(f"Unsupported data type: {data_type}")
+            raise ValueError(f"Unsupported data type: {data_type},please use DataType enum")
```
Comment (gasolin) [inline embedchain/embedchain.py:89]: nit: a space after `,`

File: embedchain/embedchain.py [hunk 2]
```diff
@@ -48,16 +49,20 @@ def _get_loader(self, data_type):
...
-        if data_type in loaders:
+        if isinstance(data_type, DataType):
             return loaders[data_type]
+        # compatible string
+        if not isinstance(data_type, DataType):
+            data_type_enum = DataType.get_enum(data_type)
+            return loaders[data_type_enum]
         else:
-            raise ValueError(f"Unsupported data type: {data_type}")
+            raise ValueError(f"Unsupported data type: {data_type},please use DataType enum")
```
Comment (gasolin) [inline embedchain/embedchain.py:65]: nit: a space after `,`

**Review by gasolin — APPROVED** [review]
👍

**Comment by guaguaguaxia** [comment]
fix it already

**Comment by aaidutta** [comment]
Hi @guaguaguaxia @gasolin i think there is value to expose the enums for all supported formats at a library level so that when someone is using the add method they can use these enums instead of writing strings.

**Comment by cachho** [comment]
I like the idea, but we need to solve merge conflicts and check if we have added new types since this PR was started.

**Comment by cachho** [comment]
We do want the data type to be self-detected at some point, but this is a good stepping stone and is still good, even if it is detected at some point.

**Comment by guaguaguaxia** [comment]
because my code behind the main branch too much,i need some time to restart the pr

**Comment by guaguaguaxia** [comment]
@cachho  hi,pls review this pr when u r free

**Pull request closed without merging**